src/thread_pool.rs

```diff
@@ -112,23 +112,27 @@ impl Registry {
 
     fn start_working(&self, index: usize) {
         log!(StartWorking { index: index });
-        let mut state = self.state.lock().unwrap();
-        state.threads_at_work += 1;
+        {
+            let mut state = self.state.lock().unwrap();
+            state.threads_at_work += 1;
+        }
         self.work_available.notify_all();
```

Comment:
Huh, I wonder why I had it in my head that one should hold the lock when notifying. It doesn't make any sense, of course. Seems good.